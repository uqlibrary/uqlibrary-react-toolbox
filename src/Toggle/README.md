# Toggle

A simple Toggle component that lists items for UQ Library applications using ReactJs

## Props
- `standard props from http://www.material-ui.com/#/components/toggle`
- `helpTitle: PropTypes.string`
- `helpText: PropTypes.any`

## Usage

**./App.js**
```jsx
import React from 'react';
...

import {Toggle} from 'uqlibrary-react-toolbox';

const App = () => (
  <div>
    <Toggle 
        name="sampleFieldName" 
        label="Toggle for Stuff"
         />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```
```
