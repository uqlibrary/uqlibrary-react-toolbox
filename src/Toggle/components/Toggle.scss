@import '../../../sass/variables';

/* thumb styles */
.toggle input + div > div:last-child > div:last-child {
    background-color: white !important;
}

/* thumb style - toggled */
.toggle input[value="true"] + div > div:last-child > div:last-child {
    background-color: $accent-color !important;
}

/* track styles */
.toggle input + div > div:last-child > div:first-child {
    background-color: $secondary-color !important;
}

/* track style - toggled */
.toggle input[value="true"] + div > div:last-child > div:first-child {
    background-color: $accent-color-light !important;
}