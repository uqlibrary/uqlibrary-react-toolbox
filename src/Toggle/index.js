export {default as Toggle} from './containers/Toggle';
