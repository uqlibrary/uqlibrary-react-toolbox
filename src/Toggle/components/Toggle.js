import React from 'react';
import PropTypes from 'prop-types';
import Toggle from 'material-ui/Toggle';

import propFilter from '../../helpers/_filterProps';
import {HelpIcon} from '../../HelpDrawer';

export default function ToggleWrapper(props) {
    const filteredProps = propFilter(props, Toggle.propTypes);
    delete filteredProps.errorText;

    return (
        <div style={{position: 'relative', width: '100%'}}>
            <Toggle {...filteredProps} />
            {props.helpText && (
                <div style={{position: 'absolute', top: '20px', right: 0}}>
                    <HelpIcon title={props.helpTitle} text={props.helpText} buttonLabel="Got it!" />
                </div>
            )}
        </div>
    );
}

ToggleWrapper.propTypes = {
    ...Toggle.propTypes,
    helpTitle: PropTypes.string,
    helpText: PropTypes.any
};
