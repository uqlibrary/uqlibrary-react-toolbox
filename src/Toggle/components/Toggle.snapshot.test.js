jest.dontMock('./Toggle');

import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Toggle from './Toggle';
import PropTypes from 'prop-types';

function setup() {
    const muiTheme = getMuiTheme();
    const props = {
        helpTitle: 'title',
        helpText: 'text',
        label: 'test'
    };

    return shallow(<Toggle {...props} />,{
        context: {muiTheme},
        childContextTypes: {muiTheme: PropTypes.object}});
}


describe('Add Toggle snapshot tests', () => {
    it('renders default toggle component', () => {
        const app = setup();
        expect(toJson(app)).toMatchSnapshot();
    });
});
