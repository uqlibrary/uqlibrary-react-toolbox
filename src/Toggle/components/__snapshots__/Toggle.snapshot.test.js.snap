// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Toggle snapshot tests renders default toggle component 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Toggle
    defaultToggled={false}
    disabled={false}
    label="test"
    labelPosition="left"
  />
  <div
    style={
      Object {
        "position": "absolute",
        "right": 0,
        "top": "20px",
      }
    }
  >
    <Connect(HelpIcon)
      buttonLabel="Ok"
      text="text"
      title="title"
    />
  </div>
</div>
`;
